centreon/src/Core/HostTemplate/Application/UseCase/FindHostTemplates/FindHostTemplates.php

```diff
@@ -0,0 +1,143 @@
+
+final class FindHostTemplates
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private readonly ReadHostTemplateRepositoryInterface $readHostTemplateRepository,
+        private readonly RequestParametersInterface $requestParameters,
+        private readonly ContactInterface $user
+    ) {
+    }
+
+    /**
+     * @param FindHostTemplatesPresenterOnPrem|FindHostTemplatesPresenterSaas $presenter
+     */
+    public function __invoke(PresenterInterface $presenter): void
+    {
+        try {
+            if (
+                ! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_TEMPLATES_READ)
+                && ! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_TEMPLATES_READ_WRITE)
+            ) {
+                $this->error(
+                    "User doesn't have sufficient rights to see host templates",
+                    ['user_id' => $this->user->getId()]
+                );
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse(HostTemplateException::accessNotAllowed())
+                );
+
+                return;
+            }
+
+            $hostTemplates = $this->readHostTemplateRepository->findByRequestParameter($this->requestParameters);
+            $presenter->present($this->createResponse($hostTemplates));
+        } catch (\Throwable $ex) {
+            $presenter->setResponseStatus(new ErrorResponse(HostTemplateException::findHostTemplates($ex)));
+            $this->error($ex->getMessage());
+        }
+    }
+
+    /**
+     * @param HostTemplate[] $hostTemplates
+     *
+     * @return FindHostTemplatesResponse
+     */
+    private function createResponse(array $hostTemplates): FindHostTemplatesResponse
+    {
+        $response = new FindHostTemplatesResponse();
+
+        foreach ($hostTemplates as $hostTemplate) {
+            $response->hostTemplates[] = [
+                'id' => $hostTemplate->getId(),
+                'name' => $hostTemplate->getName(),
+                'alias' => $hostTemplate->getAlias(),
+                'snmpVersion' => $hostTemplate->getSnmpVersion() ? $hostTemplate->getSnmpVersion()->value : null,
+                'snmpCommunity' => $hostTemplate->getSnmpCommunity(),
+                'timezoneId' => $hostTemplate->getTimezoneId(),
+                'severityId' => $hostTemplate->getSeverityId(),
+                'checkCommandId' => $hostTemplate->getCheckCommandId(),
+                'checkCommandArgs' => $hostTemplate->getCheckCommandArgs(),
+                'checkTimeperiodId' => $hostTemplate->getCheckTimeperiodId(),
+                'maxCheckAttempts' => $hostTemplate->getMaxCheckAttempts(),
+                'normalCheckInterval' => $hostTemplate->getNormalCheckInterval(),
+                'retryCheckInterval' => $hostTemplate->getretryCheckInterval(),
+                'isActiveCheckEnabled' => $hostTemplate->isActiveCheckEnabled()->toInt(),
```

Comment:
Boolean property must return a boolean value. In this case, yes, false or null but not a integer